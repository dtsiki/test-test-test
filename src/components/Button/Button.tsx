import React, { ReactNode } from 'react';
import classNames from 'classnames';

export enum ButtonVariant {
  INFO = 'is-info',
  SUCCESS = 'is-success',
  WARNING = 'is-warning',
  DANGER = 'is-danger'
}

export enum ButtonSize {
  NORMAL = 'is-normal',
  LARGE = 'is-large'
}

interface ButtonProps {
  children: ReactNode;
  variant?: ButtonVariant;
  size?: ButtonSize;
  onClick?: () => void;
}

const Button: React.FC<ButtonProps> = ({
  children,
  size = ButtonSize.NORMAL,
  variant = ButtonVariant.INFO,
  onClick
}: ButtonProps) => {
  const bind = classNames.bind([]);

  return (
    <button
      className={bind(['button', variant, size])}
      onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
