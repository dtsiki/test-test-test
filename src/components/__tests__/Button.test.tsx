import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';
import { ButtonVariant } from './../Button/Button';

const variants = Object.keys(ButtonVariant);

describe('Button', () => {
  test.each(variants)('renders %s correctly', (variant) => {
    const tree = renderer.create(<Button variant={variant as ButtonVariant}>Click me</Button>)
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });
});
