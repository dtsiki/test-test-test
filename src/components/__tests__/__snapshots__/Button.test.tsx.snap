// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders DANGER correctly 1`] = `
<button
  className="button DANGER normal"
>
  Click me
</button>
`;

exports[`Button renders INFO correctly 1`] = `
<button
  className="button INFO normal"
>
  Click me
</button>
`;

exports[`Button renders SUCCESS correctly 1`] = `
<button
  className="button SUCCESS normal"
>
  Click me
</button>
`;

exports[`Button renders WARNING correctly 1`] = `
<button
  className="button WARNING normal"
>
  Click me
</button>
`;
